libs/ledger-live-common/src/families/tron/synchronization.ts

```diff
@@ -225,4 +229,35 @@ export const postSync = (initial: TronAccount, parent: TronAccount): TronAccount
   return parent;
 };
 
+/**
+ * Merges two arrays of subAccounts according to specific rules:
+ * - The first array (subAccounts1) is up-to-date and should not be modified.
+ * - Old duplicates from the second array (subAccounts2) should be filtered out.
+ * - Only new subAccounts with a unique ID from the second array should be included.
+ * - The balance and spendableBalance fields of the second array's subAccounts should be set to 0.
+ *
+ * @param {Array} subAccounts1 - The first array of subAccounts, which is up-to-date and should not be modified.
+ * @param {Array} subAccounts2 - The second array of subAccounts, from which only new unique subAccounts should be included.
+ * @returns {Array} - The merged array of subAccounts.
+ */
+const mergeSubAccounts = (subAccounts1, subAccounts2) => {
```

Comment:
I'm scared we end up with duplicated accounts if CAL identifiers evolve (as we've seen when we moved to lowercase)